nwbinspector/nwbinspector.py

```diff
@@ -36,6 +36,68 @@ def default(self, o):
             return super().default(o)
 
 
+def configure_checks(
+    checks: list = available_checks,
+    config: Optional[dict] = None,
+    ignore: Optional[List[str]] = None,
+    select: Optional[List[str]] = None,
+    importance_threshold: Importance = Importance.BEST_PRACTICE_SUGGESTION,
+) -> list:
+    """
+    Filter a list of check functions (the entire base registry by default) according to the configuration.
+
+    Parameters
+    ----------
+    config : dict
+        Dictionary valid against our JSON configuration schema.
+        Can specify a mapping of importance levels and list of check functions whose importance you wish to change.
+        Typically loaded via json.load from a valid .json file
+    checks : list of check functions
+        Defaults to all registered checks.
+    ignore: list, optional
+        Names of functions to skip.
+    select: list, optional
+        If loading all registered checks, this can be shorthand for selecting only a handful of them.
+    importance_threshold : string, optional
+        Ignores all tests with an post-configuration assigned importance below this threshold.
+        Importance has three levels:
+            CRITICAL
+                - potentially incorrect data
+            BEST_PRACTICE_VIOLATION
+                - very suboptimal data representation
+            BEST_PRACTICE_SUGGESTION
+                - improvable data representation
+        The default is the lowest level, BEST_PRACTICE_SUGGESTION.
+    """
+    if ignore is not None and select is not None:
+        raise ValueError("Options 'ignore' and 'select' cannot both be used.")
+    if importance_threshold not in Importance:
+        raise ValueError(
+            f"Indicated importance_threshold ({importance_threshold}) is not a valid importance level! Please choose "
+            "from [CRITICAL_IMPORTANCE, BEST_PRACTICE_VIOLATION, BEST_PRACTICE_SUGGESTION]."
+        )
+
+    if config is not None:
+        checks_out = []
+        for check in checks:
+            for importance_name, func_names in config.items():
+                if check.__name__ in func_names:
+                    if importance_name == "SKIP":
+                        continue
+                    check.importance = Importance[importance_name]
+            checks_out.append(check)
+    else:
+        checks_out = checks
+
+    if select:
+        checks_out = [x for x in checks_out if x.__name__ in select]
+    elif ignore:
+        checks_out = [x for x in checks_out if x.__name__ not in ignore]
+    if importance_threshold:
+        checks_out = [x for x in checks_out if x.importance.value >= importance_threshold.value]
```

Comment:
Centralized these lines across all the calling functions so they only get called once, prior to any iteration.